### PR TITLE
Added error codes to SyntaxTest expectations

### PR DESCRIPTION
**Pull request by a3d4**
I'm not quite sure how to review it :).

### CODE REVIEW

**Comment by hrkrshnn** [comment]
What do you think about having the id after, say, TypeError? The error message would look like `TypeError 1111: (1-2): ...`

**Comment by chriseth** [comment]
I would also prefer the error id being mentioned right after the error "kind".

Review should be easy: Just run it locally and see if there is any difference.

File: test/libsolidity/SMTCheckerJSONTest.cpp [hunk 1]
```diff
@@ -127,6 +127,7 @@ TestCase::TestResult SMTCheckerJSONTest::run(ostream& _stream, string const& _li
 				end -= preamble.size();
 			m_errorList.emplace_back(SyntaxTestError{
 				error["type"].asString(),
+				nullopt,
```
Comment (a3d4) [inline test/libsolidity/SMTCheckerJSONTest.cpp]:
```
langutil::ErrorId{error["errorId"].asUInt()},
```
to enable error codes in SMT JSON tests.

File: test/libsolidity/SyntaxTest.h [hunk 2]
```diff
@@ -51,7 +51,7 @@ class SyntaxTest: public AnalysisFramework, public solidity::test::CommonSyntaxT
 protected:
 	void setupCompiler();
 	void parseAndAnalyze() override;
-	void filterObtainedErrors();
+	void filterObtainedErrors(bool withErrorIds = false);
```
Comment (a3d4) [inline test/libsolidity/SyntaxTest.h]: Revert to enable error codes in SMT tests.

File: test/CommonSyntaxTest.h [hunk 3]
```diff
@@ -33,13 +33,16 @@ namespace solidity::test
...
 	std::string type;
+	std::optional<langutil::ErrorId> errorId;
 	std::string message;
 	std::string sourceName;
 	int locationStart = -1;
 	int locationEnd = -1;
 	bool operator==(SyntaxTestError const& _rhs) const
 	{
 		return type == _rhs.type &&
+			errorId.has_value() == _rhs.errorId.has_value() &&
```
Comment (chriseth) [inline test/CommonSyntaxTest.h]: Doesn't `errorId == _rhs.errorId` do the same thing?

Comment (a3d4) [reply test/CommonSyntaxTest.h]: Yes, after adding `ErrorId::operator==` :).

**Review by chriseth — APPROVED** [review]

**Pull request merged**